client/src/main.rs

```diff
@@ -109,97 +108,292 @@ pub async fn main() -> Result<(), Box<dyn Error>> {
                         .about("gets vertex properties")
-                        .arg(&vertex_query_arg)
+                        .arg(&vertex_id_arg)
                         .arg(&optional_property_name_arg),
                 )
                 .subcommand(
                     SubCommand::with_name("edge-property")
                         .about("gets edge properties")
-                        .arg(&edge_query_arg)
+                        .args(&edge_query_arg)
                         .arg(&optional_property_name_arg),
                 ),
         )
         .subcommand(
             SubCommand::with_name("delete")
+                .setting(AppSettings::SubcommandRequiredElseHelp)
                 .subcommand(
                     SubCommand::with_name("vertex")
                         .about("deletes vertices by query")
-                        .arg(&vertex_query_arg),
+                        .arg(&vertex_id_arg),
                 )
                 .subcommand(
                     SubCommand::with_name("edge")
                         .about("deletes edges by query")
-                        .arg(&edge_query_arg)
-                        .subcommand(
-                            SubCommand::with_name("vertex-property")
-                                .about("deletes vertex properties")
-                                .arg(&vertex_query_arg)
-                                .arg(&required_property_name_arg),
-                        )
-                        .subcommand(
-                            SubCommand::with_name("edge-property")
-                                .about("deletes edge properties")
-                                .arg(&edge_query_arg)
-                                .arg(&required_property_name_arg),
-                        ),
+                        .args(&edge_query_arg),
+                )
+                .subcommand(
+                    SubCommand::with_name("vertex-property")
+                        .about("deletes vertex properties")
+                        .arg(&vertex_id_arg)
+                        .arg(&required_property_name_arg),
+                )
+                .subcommand(
+                    SubCommand::with_name("edge-property")
+                        .about("deletes edge properties")
+                        .args(&edge_query_arg)
+                        .arg(&required_property_name_arg),
                 ),
         )
         .get_matches();
 
-    let _address = matches.value_of("address").unwrap();
+    let address = matches.value_of("address").unwrap();
+    let mut client = proto::Client::new(String::from(address).try_into().unwrap())
+        .await
+        .map_err(|err| err.compat())?;
+
+    if let Some(_) = matches.subcommand_matches("ping") {
+        client.ping().await.map_err(|err| err.compat())?;
 
-    if let Some(_matches) = matches.subcommand_matches("ping") {
-        unimplemented!();
+        println!("ok");
```

Comment:
This is the only command that doesn't acquire a transaction, so to shorten the code, it might be good to restructure to something like this:

```rust
if matches.subcommand_matches("ping").is_some() {
    // ...ping
    return Ok(());
}

let trans = client.transaction().await.map_err(|err| err.compat())?;
// ...various subcommand matches here
```